libdnf/rpm/package_query.cpp

```diff
@@ -2167,32 +2167,38 @@ static void add_latest_to_map(
     }
 }
 
-static int filter_latest_sortcmp_byarch(const void * ap, const void * bp, void * dp) {
-    auto pool = static_cast<Pool *>(dp);
-    Solvable * sa = pool->solvables + *(Id *)ap;
-    Solvable * sb = pool->solvables + *(Id *)bp;
-    int r;
-    r = sa->name - sb->name;
-    if (r)
-        return r;
-    r = sa->arch - sb->arch;
-    if (r)
-        return r;
-    r = pool_evrcmp(pool, sb->evr, sa->evr, EVRCMP_COMPARE);
-    if (r)
-        return r;
-    return *(Id *)ap - *(Id *)bp;
-}
+struct LatestSolvableComparator {
+    LatestSolvableComparator(Pool * pool) : pool(pool) {}
+
+    int operator()(const Id & a, const Id & b) {
+        Solvable * sa = pool_id2solvable(pool, a);
+        Solvable * sb = pool_id2solvable(pool, b);
+
+        int r;
+        r = sa->name - sb->name;
+        if (r)
+            return r;
+        r = sa->arch - sb->arch;
+        if (r)
+            return r;
+        r = pool_evrcmp(pool, sb->evr, sa->evr, EVRCMP_COMPARE);
+        if (r)
+            return r;
+        return a - b;
+    }
+
+    Pool * pool = nullptr;
```

Comment:
Minor: `poll` is initialized in the constructor. No initialization is required here.

Reply:
To be honest I prefer the initialization here even if it's initialized in the constructor. This means if another constructor is added in the future and someone omits to set the member, it will have a known value instead of random.

Reply:
I've removed the initialization and also made the members const to make it a bit safer.